### PR TITLE
bpo-25866: Minor cleanups to "sequence" in docs

### PR DESCRIPTION
**Pull request by alclarks**
Hi, this is a pull request for my patch for BPO issue 25866.

https://bugs.python.org/issue25866

### CODE REVIEW

**Comment by brandtbucher** [comment]
Thanks @alclarks! I've recommended that #17175 be closed in favor of this PR.

File: Doc/reference/datamodel.rst [hunk 1]
```diff
@@ -1797,6 +1797,10 @@ Notes on using *__slots__*
   (the other bases must have empty slot layouts) - violations raise
   :exc:`TypeError`.
 
+* If an iterator is used for *__slots__* then a descriptor is created for each
+  of the iterator's values. However, the *__slots__* variable will be an empty
```
Comment (brandtbucher) [inline Doc/reference/datamodel.rst]:
```suggestion
  of the iterator's values. However, the *__slots__* attribute will be an empty
```

File: Doc/reference/datamodel.rst [hunk 2]
```diff
@@ -2128,8 +2132,8 @@ operators.  It is recommended that both mappings and sequences implement the
 mappings, ``in`` should search the mapping's keys; for sequences, it should
 search through the values.  It is further recommended that both mappings and
 sequences implement the :meth:`__iter__` method to allow efficient iteration
-through the container; for mappings, :meth:`__iter__` should be the same as
-:meth:`keys`; for sequences, it should iterate through the values.
+through the container; for mappings, :meth:`__iter__` should return an iterator
```
Comment (brandtbucher) [inline Doc/reference/datamodel.rst]: Minor, but in my opinion it helps the sentence read better.

```suggestion
through the container; for mappings, :meth:`__iter__` should iterate
```

**Review by brandtbucher — CHANGES_REQUESTED** [review]
A bit of minor cleanup, but otherwise this looks good:

**Comment by alclarks** [comment]
Thanks for the review @brandtbucher, does this one require a news entry?

**Comment by brandtbucher** [comment]
Do you think so? If not, I can add the label.

**Comment by alclarks** [comment]
Because it's entirely documentation and not documenting any new feature I don't think the news would be useful - but I'm happy to add one if you think it's a big enough change to warrant some news :)

**Comment by brandtbucher** [comment]
Agreed... I think your judgement is correct here.

**Comment by brandtbucher** [comment]
CC @rhettinger

**Comment by brandtbucher** [comment]
Congrats on your first CPython contribution @alclarks! :champagne:

Looking forward to seeing more from you in the future.

**Comment by alclarks** [comment]
Thanks @brandtbucher and @rhettinger for the review and merge!

**Pull request merged**